mlflow/R/mlflow/R/model.R

```diff
@@ -77,29 +77,21 @@ mlflow_predict_model <- function(model, data) {
 #' can be loaded in R. This method by default searches for a flavor supported by R/MLflow.
 #'
 #' @template roxlate-model-uri
+#' @template roxlate-client
 #' @param flavor Optional flavor specification. Can be used to load a particular flavor in case
 #'        there are multiple flavors available.
 #' @export
 mlflow_load_model <- function(model_uri, flavor = NULL, client = mlflow_client()) {
   model_path <- mlflow_download_artifacts_from_uri(model_uri, client = client)
-  supported_flavors <- supported_model_flavors()
   spec <- yaml::read_yaml(fs::path(model_path, "MLmodel"))
-  available_flavors <- intersect(names(spec$flavors), supported_flavors)
```

Comment:
Yep good question we can keep the intersection. I removed it because the old implementation ([link](https://github.com/mlflow/mlflow/pull/1269/files#diff-2aa06ff0b38d5922270e7e375aa778abL197)) of `supported_model_flavors()`, which called into `utils::methods(mlflow_)`, didn't detect flavors with private implementations of `mlflow_load_flavor`. However I'm realizing we can just return something like `append(c("keras", "crate"), old_supported_model_flavors_impl())` (i.e. hard-code the names of flavors implemented within MLflow).

Reply:
Yeah I think we need it. Without the intersection you can end up with a random flavor, could be pyfunc or spark and than you wold get a no such method exception. 

If we want to make our R flavors private, we can also concatenate with  .S3methods(mlflow_load_flavor, envir=getNamespace("mlflow")) so that we do not have to keep the list up to date manually (although tests should catch that so maybe it does not matter).